apps/dh/api-dh/source/DataHub.WebApi/Modules/ElectricityMarket/Extensions/MeasurementDtoExtensions.cs

```diff
@@ -0,0 +1,85 @@
+// Copyright 2020 Energinet DataHub A/S
+//
+// Licensed under the Apache License, Version 2.0 (the "License2");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+using Energinet.DataHub.Measurements.Abstractions.Api.Models;
+
+namespace Energinet.DataHub.WebApi.Modules.ElectricityMarket.Extensions;
+
+public static class MeasurementDtoExtensions
+{
+    public static MeasurementDto EnsureCompletePositions(this MeasurementDto measurements)
+    {
+        TimeZoneInfo centralEuropeanTimeZone;
+        try
+        {
+            centralEuropeanTimeZone = TimeZoneInfo.FindSystemTimeZoneById("Europe/Copenhagen");
+        }
+        catch
+        {
+            centralEuropeanTimeZone = TimeZoneInfo.Utc;
+        }
+
+        var resolution = DetermineResolution(measurements.MeasurementPositions);
+
+        int intervalMinutes = resolution == Resolution.QuarterHourly ? 15 : 60;
```

Comment:
For data prior to 2020, many metering points were measured monthly or yearly

Reply:
Do you have any metering point ids with these examples?

Reply:
Lets see how it looks